home.php

```diff
@@ -0,0 +1,166 @@
+    unset($_GET['_ROUTE']);
+    $_SERVER['QUERY_STRING'] = http_build_query($_GET);
+
+    // Fix __FILE__ and __DIR__ constants by updating $_SERVER superglobals
+    // This ensures target files see correct paths in their execution context
+    $_SERVER['SCRIPT_FILENAME'] = $targetFile;
+    $_SERVER['SCRIPT_NAME'] = '/' . $route;
+    $_SERVER['PHP_SELF'] = '/' . $route;
+
+    // Change to target file's directory to preserve relative path behavior
+    chdir(dirname($targetFile));
+
+    // Include target file - it now executes with correct context
+    require $targetFile;
+    exit;
+}
+
+// Early extension point via event system
+// Modules can listen to FrontControllerEvent::EVENT_EARLY for early initialization
+// See https://github.com/adunsulag/oe-module-custom-skeleton for module development
+if (isset($GLOBALS['kernel']) && method_exists($GLOBALS['kernel'], 'getEventDispatcher')) {
+    $event = new FrontControllerEvent($_GET['_ROUTE'] ?? '', $_GET['site'] ?? 'default');
+    $GLOBALS['kernel']->getEventDispatcher()->dispatch($event, FrontControllerEvent::EVENT_EARLY);
+}
+
+// Initialize router
+$router = new Router(__DIR__);
+
+// Determine multisite and route
+$router->determineSiteId();
+$route = $router->extractRoute();
+
+// Handle trailing slash
+$router->handleTrailingSlash();
+
+// Deny forbidden paths
+if ($router->isForbiddenPath()) {
+    http_response_code(404);
+    SecurityValidator::logSecurityEvent("Denied path: $route");
+    exit('Not Found');
+}
+
+// Admin-only paths
+if ($router->requiresAdmin()) {
+    $_SERVER['REQUIRE_ADMIN'] = true;
+}
+
+// Block .inc.php files
+if (SecurityValidator::isIncludeFile($route)) {
+    http_response_code(403);
+    SecurityValidator::logSecurityEvent("Blocked .inc.php access: $route");
+    exit(
+        SecurityValidator::getDebugMessage(
+            'Access Denied',
+            'Access Denied: Include files cannot be accessed directly'
+        )
+    );
+}
+
+// Resolve and validate target file
+$targetFile = $router->resolveTargetFile();
+
+if ($targetFile === null) {
+    http_response_code(404);
+    SecurityValidator::logSecurityEvent("Invalid or non-existent path: $route");
+    exit('Not Found');
+}
+
+// Route to target file with error handling and shutdown function registration
+// Register shutdown function to dispatch late event even if target file exits early
+register_shutdown_function(function () use ($route, $router) {
+    if (isset($GLOBALS['kernel']) && method_exists($GLOBALS['kernel'], 'getEventDispatcher')) {
+        $error = error_get_last();
+        $context = [
+            'completed_normally' => $error === null || !in_array($error['type'], [E_ERROR, E_PARSE, E_CORE_ERROR, E_COMPILE_ERROR]),
+            'error' => $error,
+        ];
+        $event = new FrontControllerEvent($route, $router->getSiteId(), $context);
+        $GLOBALS['kernel']->getEventDispatcher()->dispatch($event, FrontControllerEvent::EVENT_LATE);
+    }
+});
```

Comment:
Would it make sense to create the shutdown function once and then use it both here and below on line 155?